src/wtmp.c

```diff
@@ -40,7 +40,7 @@ int log_wtmp(const char *line, const char *name, const char *host,
     !(defined(LINUX) || defined(__hpux) || defined (_AIX))
   /* This "auxilliary" utmp doesn't exist under linux. */
 
-#if defined(__sparcv9) && !defined(__NetBSD__) && !defined(__FreeBSD__)
+#if defined(__sun)
```

Comment:
So then there are _two_ goals for this change: 1) make code compile properly on Solaris, and 2) restrict this code path to just Solaris?  I'm OK with the first purpose/goal, but would rather spend the time to rework the Autoconf detection if the second goal is required.